src/hooks/useDraftSpecEditor.ts

```diff
@@ -3,64 +3,86 @@ import { AllowedScopes } from 'components/editor/MonacoEditor/types';
 import {
     useEditorStore_currentCatalog,
     useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_draftSpecs,
+    useEditorStore_queryResponse_isValidating,
+    useEditorStore_queryResponse_mutate,
     useEditorStore_setSpecs,
 } from 'components/editor/Store/hooks';
-import useDraftSpecs, { DraftSpec } from 'hooks/useDraftSpecs';
-import { useEffect, useState } from 'react';
+import { DraftSpec } from 'hooks/useDraftSpecs';
+import { get, has, set } from 'lodash';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { stringifyJSON } from 'services/stringify';
 import { Entity } from 'types';
 
 function useDraftSpecEditor(
     entityName: string | undefined,
-    entityType: Entity,
-    useLocalScope: boolean
+    localScope?: boolean,
+    editorSchemaScope?: AllowedScopes | string
 ) {
     // Local State
     const [draftSpec, setDraftSpec] = useState<DraftSpec>(null);
 
     // Draft Editor Store
     const currentCatalog = useEditorStore_currentCatalog({
-        localScope: useLocalScope,
+        localScope,
     });
     const setSpecs = useEditorStore_setSpecs({
-        localScope: useLocalScope,
+        localScope,
     });
     const draftId = useEditorStore_persistedDraftId();
 
-    // Fetch the draft specs for this draft
-    const { draftSpecs, isValidating, mutate } = useDraftSpecs(draftId, {
-        specType: entityType,
-        catalogName: entityName,
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs({ localScope });
```

Comment:
Not for this pr but with the query in the store already we should really stop duplicating these into the `specs` as it does not seem to solve any problems.

Reply:
As discussed, there will be a PR dedicated to overhauling the editor store. This will be taken care of at that time.